commands/utils/quick-pick-azure.ts

```diff
@@ -0,0 +1,168 @@
+    let desiredReg: IAzureQuickPickItem<RegistryWrapper> = await ext.ui.showQuickPick(quickPickRegList, {
+        'canPickMany': false,
+        'placeHolder': placeHolder
+    });
+    let registry: Registry;
+    if (canCreateNew && desiredReg.data.isCreateNew) {
+        registry = <Registry>await vscode.commands.executeCommand("vscode-docker.create-ACR-Registry");
+    } else {
+        registry = desiredReg.data.registry;
+    }
+    return registry;
+}
+
+export async function quickPickSKU(): Promise<string> {
+    const quickPickSkuList = skus.map(sk => <IAzureQuickPickItem<string>>{ label: sk, data: sk });
+    let desiredSku: IAzureQuickPickItem<string> = await ext.ui.showQuickPick(quickPickSkuList, { 'canPickMany': false, 'placeHolder': 'Choose a SKU to use' });
+    return desiredSku.data;
+}
+
+export async function quickPickSubscription(): Promise<Subscription> {
+    const subscriptions = AzureUtilityManager.getInstance().getFilteredSubscriptionList();
+    if (subscriptions.length === 0) {
+        vscode.window.showErrorMessage("You do not have any subscriptions. You can create one in your Azure portal", "Open Portal").then(val => {
+            if (val === "Open Portal") {
+                opn('https://portal.azure.com/');
+            }
+        });
+        throw new Error('User has no azure subscriptions');
+    }
+    let quickPickSubList = subscriptions.map(sub => <IAzureQuickPickItem<Subscription>>{ label: sub.displayName, data: sub });
+    let desiredSub = await ext.ui.showQuickPick(quickPickSubList, {
+        'canPickMany': false,
+        'placeHolder': 'Choose a subscription to use'
+    });
+    return desiredSub.data;
+}
+
+export async function quickPickLocation(subscription: Subscription): Promise<string> {
+    let locations: Location[] = await AzureUtilityManager.getInstance().getLocationsBySubscription(subscription);
+    let quickPickLocList = locations.map(loc => <IAzureQuickPickItem<Location>>{ label: loc.displayName, data: loc });
+
+    quickPickLocList.sort((loc1, loc2): number => {
+        return loc1.data.displayName.localeCompare(loc2.data.displayName);
+    });
+
+    let desiredLocation: IAzureQuickPickItem<Location> = await ext.ui.showQuickPick(quickPickLocList, {
+        'canPickMany': false,
+        'placeHolder': 'Choose a location to use'
+    });
+    return desiredLocation.label;
+}
+
+export async function quickPickResourceGroup(canCreateNew?: boolean, subscription?: Subscription): Promise<ResourceGroup> {
+    let resourceGroups = await AzureUtilityManager.getInstance().getResourceGroups(subscription);
+    let resourceGroupNames: string[] = [];
+
+    if (canCreateNew) { resourceGroupNames.push('+ Create new resource group'); }
+    for (let resGroupName of resourceGroups) {
+        resourceGroupNames.push(resGroupName.name);
+    }
+
+    let resourceGroupName = await vscode.window.showQuickPick(resourceGroupNames, {
```

Comment:
ext.ui.showQuickPick().

With your current code, the user can press cancel on this step, but the code keeps going and eventually ends up with "Error: Cannot read property 'name' of undefined"